tests/submit_tests/test_utils.py

```diff
@@ -17,3 +18,37 @@ def test_calculate_past_date(submit_test_client):
 
     days_remaining = days_between_dates(date_1, date_2)
     assert days_remaining == -20
+
+
+def test_ascii_safe_filename_with_non_ascii_characters():
+    filename = "téstfilé_2024.xlsx"
+    expected_output = "t--stfil--_2024.xlsx"
+
+    assert make_ascii_safe_filename(filename) == expected_output
+
+
+def test_ascii_safe_filename_with_trailing_non_ascii():
+    filename = "testfile_2024é.xlsx"
+    expected_output = "testfile_2024--.xlsx"
+
+    assert make_ascii_safe_filename(filename) == expected_output
+
+
+def test_ascii_safe_filename_with_empty_filename():
+    filename = ""
+    expected_output = ""
+
+    assert make_ascii_safe_filename(filename) == expected_output
+
+
+def test_ascii_safe_filename_with_invalid_ascii_characters():
+    filename = "file_2024é.xlsx"
+    expected_output = "file_2024--.xlsx"
+
+    assert make_ascii_safe_filename(filename) == expected_output
+
+
+def test_ascii_safe_filename_with_only_invalid_ascii():
+    filename = "file_2024\u2019.xlsx"
+    expected_output = "file_2024--.xlsx"
+    assert make_ascii_safe_filename(filename) == expected_output
```

Comment:
Where the logic of the test is the same, just the values are different, it can be a lot cleaner and simpler to use test parameterisation instead:

```suggestion
@pytest.mark.parametrize(
    'filename, expected_output',
    (
        ("téstfilé_2024.xlsx", "t--stfil--_2024.xlsx"),
        ("testfile_2024é.xlsx", "testfile_2024_.xlsx"),
        ("file_2024é.xlsx", "file_2024--.xlsx"),
        ("file_2024\u2019.xlsx", "file_2024--.xlsx"),
    )
)
def test_ascii_safe_filename(filename, expected_output):
    assert make_ascii_safe_filename(filename) == expected_output
```

Reply:
Done!